dkg/dkg.go

```diff
@@ -48,6 +48,55 @@ type Config struct {
 	TestSigning bool
 }
 
+// exchanger is responsible for exchanging partial signatures between peers on libp2p.
+type exchanger struct {
+	setChan chan core.ParSignedDataSet
+	sigex   *parsigex.ParSigEx
+}
+
+func newExchanger(tcpNode host.Host, peerIdx int, peers []peer.ID) exchanger {
+	ex := exchanger{
+		sigex:   parsigex.NewParSigEx(tcpNode, peerIdx, peers),
+		setChan: make(chan core.ParSignedDataSet, len(peers)),
+	}
+
+	// Wiring core workflow components
+	ex.sigex.Subscribe(ex.getSets)
+
+	return ex
+}
+
+func (e *exchanger) exchange(ctx context.Context, set core.ParSignedDataSet) ([]core.ParSignedDataSet, error) {
```

Comment:
godoc FTW. it isn't obvious what `[]core.ParSignedDataSet` is...

Reply:
you mean to be specific about whether it is lockhash or deposit data ?